src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -63,15 +63,25 @@ public static void ValidateValue(ReadOnlySpan<byte> value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void ValidateDouble(double value)
         {
+#if !netstandard
             if (!double.IsFinite(value))
                 ThrowHelper.ThrowArgumentException_ValueNotSupported();
+#else
+            if (double.IsNaN(value) || double.IsInfinity(value))
+                ThrowHelper.ThrowArgumentException_ValueNotSupported();
+#endif
```

Comment:
You could avoid duplicating the ThrowHelper line, e.g.
```C#
#if !netstandard
    if (!double.IsFinite(value))
#else
    if (double.IsNan(value) || double.IsInfinity(value))
#endif
        ThrowHelper.ThrowArgumentException_ValueNotSupported();
```

Reply:
In that case, I would add back the braces to avoid confusion.